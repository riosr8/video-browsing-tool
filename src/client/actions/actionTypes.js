export const LOGIN_START = 'LOGIN_START';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const LOGIN_FAIL = 'LOGIN_FAIL';
export const AUTH_LOGOUT = 'AUTH_LOGOUT';

export const REGISTRATION_START = 'REGISTRATION_START';
export const REGISTRATION_SUCCESS = 'REGISTRATION_SUCCESS';
export const REGISTRATION_FAIL = 'REGISTRATION_FAIL';
export const RESET_REGISTRATION_COMPLETED = 'RESET_REGISTRATION_COMPLETED';

export const BEGIN_FORM_ERROR_CHECKING = 'BEGIN_FORM_ERROR_CHECKING';
export const SET_FORM_ERROR_START = 'SET_FORM_ERROR_START';
export const SET_FORM_ERROR_END = 'SET_FORM_ERROR_END';

export const SET_AUTH_REDIRECT_PATH = 'SET_AUTH_REDIRECT_PATH';

export const FETCH_VIDEOS_START = 'FETCH_VIDEOS_START';
export const FETCH_VIDEOS_SUCCESS = 'FETCH_VIDEOS_SUCCESS';
export const FETCH_VIDEOS_FAIL = 'FETCH_VIDEOS_FAIL';

export const UPLOAD_VIDEO_START = 'UPLOAD_VIDEO_START';
export const UPLOAD_VIDEO_SUCCESS = 'UPLOAD_VIDEO_SUCCESS';
export const UPLOAD_VIDEO_FAIL = 'UPLOAD_VIDEO_FAIL';

export const DELETE_VIDEO_START = 'DELETE_VIDEO_START';
export const DELETE_VIDEO_SUCCESS = 'DELETE_VIDEO_SUCCESS';
export const DELETE_VIDEO_FAIL = 'DELETE_VIDEO_FAIL';

export const REQUEST_PROCESSING_START = 'REQUEST_PROCESSING_START';
export const REQUEST_PROCESSING_SUCCESS = 'REQUEST_PROCESSING_SUCCESS';
export const REQUEST_PROCESSING_FAIL = 'REQUEST_PROCESSING_FAIL';

export const FETCH_ALGORITHMS_START = 'FETCH_ALGORITHMS_START';
export const FETCH_ALGORITHMS_SUCCESS = 'FETCH_ALGORITHMS_SUCCESS';
export const FETCH_ALGORITHMS_FAIL = 'FETCH_ALGORITHMS_FAIL';

export const FETCH_RESULTS_START = 'FETCH_RESULTS_START';
export const FETCH_RESULTS_SUCCESS = 'FETCH_RESULTS_SUCCESS';
export const FETCH_RESULTS_FAIL = 'FETCH_RESULTS_FAIL';

export const POLL_STATUS_START = 'POLL_STATUS_START';
export const POLL_STATUS_SUCCESS = 'POLL_STATUS_SUCCESS';
export const POLL_STATUS_FAIL = 'POLL_STATUS_FAIL';

export const CLEAR_MESSAGE = 'CLEAR_MESSAGE';

export const SET_ACTIVE_PAGE = 'SET_ACTIVE_PAGE';
export const SET_SORT_KEY = 'SET_SORT_KEY';
export const SET_SORT_ORDER = 'SET_SORT_ORDER';
