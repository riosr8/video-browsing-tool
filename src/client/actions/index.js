export {
  auth,
  setAuthRedirectPath,
  checkAuthState,
  logout,
  register,
  setError,
  beginFormErrorChecking,
  resetRegistrationCompleted,
} from './auth';

export {
  getVideos,
  uploadVideo,
  sendProcessingRequest,
  deleteVideo,
  getAlgorithms,
  getResults,
  clearMessage,
  setActivePage,
  setSortKey,
  setSortOrder,
} from './dashboard';

export { pollStatus } from './processStatus';
