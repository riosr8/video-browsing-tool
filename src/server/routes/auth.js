const express = require('express');

const router = express.Router();
const jwt = require('jsonwebtoken');
const passport = require('passport');
const bodyParser = require('body-parser');

const config = require('../config/config');
// eslint-disable-next-line
router.post('/login', bodyParser.json(), (req, res, next) => {
  // eslint-disable-next-line
  passport.authenticate('local', { session: false }, (err, user, info) => {
    if (err || !user) {
      if (err === null) {
        return res.status(400).json({
          message: info.message,
        });
      }
      return res.status(400).json({
        message: err.message,
      });
    }
    req.login(user, { session: false }, error => {
      if (error) {
        res.send(error);
      }
      const token = jwt.sign(
        JSON.stringify({ email: user.email, password: user.password }),
        config.authenticationSecret.secret,
      );
      res.status(200);
      return res.json({ email: user.email, token, expiresIn: 3600000 });
    });
  })(req, res);
});

module.exports = router;
