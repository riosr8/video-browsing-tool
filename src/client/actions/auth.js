import axios from 'axios';
import * as actionTypes from './actionTypes';

export const loginStart = () => {
  return {
    type: actionTypes.LOGIN_START,
  };
};

export const loginSuccess = (token, userId) => {
  return {
    type: actionTypes.LOGIN_SUCCESS,
    idToken: token,
    userId,
  };
};

export const loginFail = error => {
  return {
    type: actionTypes.LOGIN_FAIL,
    error,
  };
};

export const registrationStart = () => {
  return {
    type: actionTypes.REGISTRATION_START,
  };
};

export const registrationSuccess = () => {
  return {
    type: actionTypes.REGISTRATION_SUCCESS,
  };
};

export const registrationFail = error => {
  return {
    type: actionTypes.REGISTRATION_FAIL,
    error,
  };
};

export const logout = () => {
  localStorage.removeItem('token');
  localStorage.removeItem('expirationDate');
  localStorage.removeItem('userId');
  return {
    type: actionTypes.AUTH_LOGOUT,
  };
};

export const beginFormErrorChecking = () => {
  return {
    type: actionTypes.BEGIN_FORM_ERROR_CHECKING,
  };
};

export const setFormErrorStart = () => {
  return {
    type: actionTypes.SET_FORM_ERROR_START,
  };
};

export const setFormErrorEnd = error => {
  return {
    type: actionTypes.SET_FORM_ERROR_END,
    error,
  };
};

export const resetRegistrationCompleted = () => {
  return {
    type: actionTypes.RESET_REGISTRATION_COMPLETED,
  };
};

export const setError = error => {
  return dispatch => {
    dispatch(setFormErrorStart());
    dispatch(setFormErrorEnd(error));
  };
};

export const auth = (email, password) => {
  return async dispatch => {
    dispatch(loginStart());
    const authData = {
      email,
      password,
    };
    await axios
      .post('/auth/login', authData)
      .then(response => {
        const expirationDate = new Date(new Date().getTime() + response.data.expiresIn);
        localStorage.setItem('token', response.data.token);
        localStorage.setItem('expirationDate', expirationDate);
        localStorage.setItem('userId', response.data.email);
        dispatch(loginSuccess(response.data.token, response.data.email));
      })
      .catch(err => {
        dispatch(loginFail(err.response.data.message));
      });
  };
};

export const register = (email, password) => {
  return async dispatch => {
    dispatch(registrationStart());
    const registrationData = {
      email,
      password,
    };
    await axios
      .post('/register/newUser', registrationData)
      // eslint-disable-next-line no-unused-vars
      .then(response => {
        dispatch(registrationSuccess());
      })
      .catch(err => {
        dispatch(registrationFail(err.response.message));
      });
  };
};

export const setAuthRedirectPath = path => {
  return {
    type: actionTypes.SET_AUTH_REDIRECT_PATH,
    path,
  };
};

export const checkAuthTimeout = expirationTime => {
  return dispatch => {
    setTimeout(() => {
      dispatch(logout());
    }, expirationTime * 1000);
  };
};

export const checkAuthState = () => {
  return dispatch => {
    const token = localStorage.getItem('token');
    if (!token) {
      dispatch(logout());
    } else {
      const expirationDate = new Date(localStorage.getItem('expirationDate'));
      if (expirationDate <= new Date()) {
        dispatch(logout());
      } else {
        const userId = localStorage.getItem('userId');
        dispatch(loginSuccess(token, userId));
        dispatch(checkAuthTimeout((expirationDate.getTime() - new Date().getTime()) / 1000));
      }
    }
  };
};
