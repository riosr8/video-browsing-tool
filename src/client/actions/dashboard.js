import axios from 'axios';
import * as actionTypes from './actionTypes';

// FETCH VIDEO ACTION CREATORS

export const fetchVideosStart = () => {
  return {
    type: actionTypes.FETCH_VIDEOS_START,
  };
};

export const fetchVideosSuccess = (userVideos, totalPages) => {
  return {
    type: actionTypes.FETCH_VIDEOS_SUCCESS,
    userVideos,
    totalPages,
  };
};

export const fetchVideosFail = error => {
  return {
    type: actionTypes.FETCH_VIDEOS_FAIL,
    error,
  };
};

export const uploadVideoStart = () => {
  return {
    type: actionTypes.UPLOAD_VIDEO_START,
  };
};

export const uploadVideoSuccess = success => {
  return {
    type: actionTypes.UPLOAD_VIDEO_SUCCESS,
    success,
  };
};

export const uploadVideoFail = error => {
  return {
    type: actionTypes.UPLOAD_VIDEO_FAIL,
    error,
  };
};

export const requestProcessingStart = () => {
  return {
    type: actionTypes.REQUEST_PROCESSING_START,
  };
};

export const requestProcessingSuccess = message => {
  return {
    type: actionTypes.REQUEST_PROCESSING_SUCCESS,
    success: message,
  };
};

export const requestProcessingFail = error => {
  return {
    type: actionTypes.REQUEST_PROCESSING_FAIL,
    error,
  };
};

export const fetchAlgorithmsStart = () => {
  return {
    type: actionTypes.FETCH_ALGORITHMS_START,
  };
};

export const fetchAlgorithmsSuccess = algorithms => {
  return {
    type: actionTypes.FETCH_ALGORITHMS_SUCCESS,
    algorithms,
  };
};

export const fetchAlgorithmsFail = () => {
  return {
    type: actionTypes.FETCH_ALGORITHMS_FAIL,
  };
};

export const deleteVideoStart = () => {
  return {
    type: actionTypes.DELETE_VIDEO_START,
  };
};

export const deleteVideoSuccess = success => {
  return {
    type: actionTypes.DELETE_VIDEO_SUCCESS,
    success,
  };
};

export const deleteVideoFail = error => {
  return {
    type: actionTypes.DELETE_VIDEO_FAIL,
    error,
  };
};

export const fetchResultsStart = () => {
  return {
    type: actionTypes.FETCH_RESULTS_START,
  };
};

export const fetchResultsSuccess = results => {
  return {
    type: actionTypes.FETCH_RESULTS_SUCCESS,
    results,
  };
};

export const fetchResultsFail = error => {
  return {
    type: actionTypes.FETCH_RESULTS_FAIL,
    error,
  };
};

export const clearMessage = () => {
  return {
    type: actionTypes.CLEAR_MESSAGE,
  };
};

export const setActivePage = activePage => {
  return {
    type: actionTypes.SET_ACTIVE_PAGE,
    activePage,
  };
};

export const setSortKey = sortKey => {
  return {
    type: actionTypes.SET_SORT_KEY,
    sortKey,
  };
};

export const setSortOrder = sortOrder => {
  return {
    type: actionTypes.SET_SORT_ORDER,
    sortOrder,
  };
};

export const getResults = (token, uuid) => {
  return async dispatch => {
    dispatch(fetchResultsStart());
    await axios
      .post(
        '/api/results',
        {
          videoUuid: uuid,
        },
        {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        },
      )
      .then(res => {
        dispatch(fetchResultsSuccess(res.data));
      })
      .catch(err => {
        dispatch(fetchResultsFail(err.response.data.message));
      });
  };
};

export const getVideos = (token, pageNo, sortKey, sortOrder) => {
  return async dispatch => {
    dispatch(fetchVideosStart());
    await axios
      .get(`/api/listPagination?pageNo=${pageNo}&sortVal=${sortKey}&order=${sortOrder}`, {
        headers: {
          'Access-Control-Allow-Origin': '*',
          'Access-Control-Allow-Methods': 'GET,PUT,POST,DELETE,PATCH,OPTIONS',
          Authorization: `Bearer ${token}`,
        },
      })
      .then(res => {
        dispatch(fetchVideosSuccess(res.data.results, res.data.totalCount));
      })
      .catch(err => {
        dispatch(fetchVideosFail(err.response.data.message));
      });
  };
};

export const getAlgorithms = () => {
  return async dispatch => {
    dispatch(fetchAlgorithmsStart());
    await axios
      .get('/noAuth/algorithmsAvailable')
      .then(res => {
        dispatch(fetchAlgorithmsSuccess(res.data.algorithms));
      })
      .catch(err => {
        dispatch(fetchAlgorithmsFail(err.response.data.message));
      });
  };
};

export const uploadVideo = (data, token, activePage, sortKey, sortOrder) => {
  return async dispatch => {
    dispatch(uploadVideoStart());
    await axios
      .post('/api/upload', data, {
        headers: {
          'Access-Control-Allow-Origin': '*',
          'Access-Control-Allow-Methods': 'GET,PUT,POST,DELETE,PATCH,OPTIONS',
          Authorization: `Bearer ${token}`,
        },
      })
      .then(res => {
        dispatch(uploadVideoSuccess(`${res.data.videoName} successfully uploaded.`));
      })
      .then(() => {
        dispatch(getVideos(token, activePage, sortKey, sortOrder));
      })
      .catch(err => {
        dispatch(uploadVideoFail(err.response.data.message));
      });
  };
};

export const sendProcessingRequest = (token, selectedAlgorithms, selectedVideoUUID) => {
  return async dispatch => {
    dispatch(requestProcessingStart());
    await axios
      .post(
        '/api/algorithmInput',
        {
          videoUuid: selectedVideoUUID,
          algorithmicInput: selectedAlgorithms,
        },
        {
          headers: {
            'Access-Control-Allow-Origin': '*',
            'Access-Control-Allow-Methods': 'GET,PUT,POST,DELETE,PATCH,OPTIONS',
            Authorization: `Bearer ${token}`,
          },
        },
      )
      .then(res => {
        dispatch(requestProcessingSuccess(res.data.message));
      })
      .catch(err => {
        dispatch(requestProcessingFail(err.response.data.message));
      });
  };
};

export const deleteVideo = (token, uuid, activePage, sortKey, sortOrder) => {
  return async dispatch => {
    dispatch(deleteVideoStart());
    await axios
      .post(
        '/api/delete',
        {
          videoUuid: uuid,
        },
        {
          headers: {
            'Access-Control-Allow-Origin': '*',
            'Access-Control-Allow-Methods': 'GET,PUT,POST,DELETE,PATCH,OPTIONS',
            Authorization: `Bearer ${token}`,
          },
        },
      )
      .then(res => {
        dispatch(deleteVideoSuccess(res.data.message));
      })
      .then(() => {
        dispatch(getVideos(token, activePage, sortKey, sortOrder));
      })
      .catch(err => {
        dispatch(deleteVideoFail(err.response.data.message));
      });
  };
};
