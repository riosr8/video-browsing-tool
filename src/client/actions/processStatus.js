import axios from 'axios';
import * as actionTypes from './actionTypes';

export const pollStatusStart = () => {
  return {
    type: actionTypes.POLL_STATUS_START,
  };
};

export const pollStatusSuccess = status => {
  return {
    type: actionTypes.POLL_STATUS_SUCCESS,
    status,
  };
};

export const pollStatusFail = error => {
  return {
    type: actionTypes.POLL_STATUS_FAIL,
    error,
  };
};

export const pollStatus = token => {
  return async dispatch => {
    dispatch(pollStatusStart());
    await axios
      .get('/api/status', {
        headers: {
          'Access-Control-Allow-Origin': '*',
          'Access-Control-Allow-Methods': 'GET,PUT,POST,DELETE,PATCH,OPTIONS',
          Authorization: `Bearer ${token}`,
        },
      })
      .then(res => {
        dispatch(pollStatusSuccess(res.data));
      })
      .catch(err => {
        dispatch(pollStatusFail(err.response.data.message));
      });
  };
};
