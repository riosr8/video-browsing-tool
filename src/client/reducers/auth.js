import * as actionTypes from '../actions/actionTypes';
import { updateObject } from '../util/util';

export const initialState = {
  token: null,
  userId: null,
  error: null,
  loading: false,
  registering: false,
  registrationComplete: false,
  authRedirectPath: '/',
};
const authStart = state => updateObject(state, { error: null, loading: true });

const authSuccess = (state, action) => {
  return updateObject(state, {
    token: action.idToken,
    userId: action.userId,
    error: null,
    loading: false,
  });
};

const authFail = (state, action) => {
  return updateObject(state, {
    error: action.error,
    loading: false,
  });
};

const authLogout = state => {
  return updateObject(state, { token: null, userId: null });
};

const beginFormErrorChecking = state => {
  return updateObject(state, {
    error: null,
  });
};

const setFormErrorStart = state => {
  return updateObject(state, {
    error: null,
  });
};

const setFormErrorEnd = (state, action) => {
  return updateObject(state, {
    error: action.error,
  });
};

const registrationStart = state => {
  return updateObject(state, {
    error: null,
    loading: true,
    registering: true,
    registrationComplete: false,
  });
};

const registrationSuccess = state => {
  return updateObject(state, {
    error: null,
    loading: false,
    registering: false,
    registrationComplete: true,
  });
};

const registrationFail = (state, action) => {
  return updateObject(state, {
    error: action.error,
    loading: false,
    registering: false,
    registrationComplete: false,
  });
};

const resetRegistrationCompleted = state => {
  return updateObject(state, {
    registrationComplete: false,
  });
};

const setAuthRedirectPath = (state, action) => {
  return updateObject(state, { authRedirectPath: action.path });
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case actionTypes.LOGIN_START:
      return authStart(state);
    case actionTypes.LOGIN_SUCCESS:
      return authSuccess(state, action);
    case actionTypes.LOGIN_FAIL:
      return authFail(state, action);
    case actionTypes.REGISTRATION_START:
      return registrationStart(state);
    case actionTypes.REGISTRATION_SUCCESS:
      return registrationSuccess(state);
    case actionTypes.REGISTRATION_FAIL:
      return registrationFail(state, action);
    case actionTypes.RESET_REGISTRATION_COMPLETED:
      return resetRegistrationCompleted(state);
    case actionTypes.AUTH_LOGOUT:
      return authLogout(state);
    case actionTypes.BEGIN_FORM_ERROR_CHECKING:
      return beginFormErrorChecking(state);
    case actionTypes.SET_FORM_ERROR_START:
      return setFormErrorStart(state);
    case actionTypes.SET_FORM_ERROR_END:
      return setFormErrorEnd(state, action);
    case actionTypes.SET_AUTH_REDIRECT_PATH:
      return setAuthRedirectPath(state, action);
    default:
      return state;
  }
};

export default reducer;
