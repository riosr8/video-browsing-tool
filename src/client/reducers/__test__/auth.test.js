import reducer, { initialState } from '../auth';
import * as actions from '../../actions/auth';

const tests = [
  { name: 'should handle LOGIN_START', action: actions.loginStart },
  {
    name: 'should handle LOGIN_SUCCESS',
    action: () => {
      return actions.loginSuccess('token', 'userid');
    },
  },
  {
    name: 'should handle LOGIN_FAIL',
    action: () => {
      return actions.loginFail('fail');
    },
  },
  { name: 'should handle AUTH_LOGOUT', action: actions.logout },
  { name: 'should handle REGISTRATION_START', action: actions.registrationStart },
  { name: 'should handle REGISTRATION_SUCCESS', action: actions.registrationSuccess },
  {
    name: 'should handle REGISTRATION_FAIL',
    action: () => {
      return actions.registrationFail('fail');
    },
  },
  {
    name: 'should handle RESET_REGISTRATION_COMPLETED',
    action: actions.resetRegistrationCompleted,
  },
  { name: 'should handle BEGIN_FORM_ERROR_CHECKING', action: actions.beginFormErrorChecking },
  { name: 'should handle SET_FORM_ERROR_START', action: actions.setFormErrorStart },
  {
    name: 'should handle SET_FORM_ERROR_END',
    action: () => {
      return actions.setFormErrorEnd('fail');
    },
  },
  {
    name: 'should handle SET_AUTH_REDIRECT_PATH',
    action: () => {
      return actions.setAuthRedirectPath('/');
    },
  },
];

describe('Auth reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  tests.forEach(test => {
    it(test.name, () => {
      expect(reducer(initialState, test.action())).toMatchSnapshot();
    });
  });
});
