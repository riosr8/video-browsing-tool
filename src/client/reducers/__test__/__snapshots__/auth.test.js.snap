// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth reducer should handle AUTH_LOGOUT 1`] = `
Object {
  "authRedirectPath": "/",
  "error": null,
  "loading": false,
  "registering": false,
  "registrationComplete": false,
  "token": null,
  "userId": null,
}
`;

exports[`Auth reducer should handle BEGIN_FORM_ERROR_CHECKING 1`] = `
Object {
  "authRedirectPath": "/",
  "error": null,
  "loading": false,
  "registering": false,
  "registrationComplete": false,
  "token": null,
  "userId": null,
}
`;

exports[`Auth reducer should handle LOGIN_FAIL 1`] = `
Object {
  "authRedirectPath": "/",
  "error": "fail",
  "loading": false,
  "registering": false,
  "registrationComplete": false,
  "token": null,
  "userId": null,
}
`;

exports[`Auth reducer should handle LOGIN_START 1`] = `
Object {
  "authRedirectPath": "/",
  "error": null,
  "loading": true,
  "registering": false,
  "registrationComplete": false,
  "token": null,
  "userId": null,
}
`;

exports[`Auth reducer should handle LOGIN_SUCCESS 1`] = `
Object {
  "authRedirectPath": "/",
  "error": null,
  "loading": false,
  "registering": false,
  "registrationComplete": false,
  "token": "token",
  "userId": "userid",
}
`;

exports[`Auth reducer should handle REGISTRATION_FAIL 1`] = `
Object {
  "authRedirectPath": "/",
  "error": "fail",
  "loading": false,
  "registering": false,
  "registrationComplete": false,
  "token": null,
  "userId": null,
}
`;

exports[`Auth reducer should handle REGISTRATION_START 1`] = `
Object {
  "authRedirectPath": "/",
  "error": null,
  "loading": true,
  "registering": true,
  "registrationComplete": false,
  "token": null,
  "userId": null,
}
`;

exports[`Auth reducer should handle REGISTRATION_SUCCESS 1`] = `
Object {
  "authRedirectPath": "/",
  "error": null,
  "loading": false,
  "registering": false,
  "registrationComplete": true,
  "token": null,
  "userId": null,
}
`;

exports[`Auth reducer should handle RESET_REGISTRATION_COMPLETED 1`] = `
Object {
  "authRedirectPath": "/",
  "error": null,
  "loading": false,
  "registering": false,
  "registrationComplete": false,
  "token": null,
  "userId": null,
}
`;

exports[`Auth reducer should handle SET_AUTH_REDIRECT_PATH 1`] = `
Object {
  "authRedirectPath": "/",
  "error": null,
  "loading": false,
  "registering": false,
  "registrationComplete": false,
  "token": null,
  "userId": null,
}
`;

exports[`Auth reducer should handle SET_FORM_ERROR_END 1`] = `
Object {
  "authRedirectPath": "/",
  "error": "fail",
  "loading": false,
  "registering": false,
  "registrationComplete": false,
  "token": null,
  "userId": null,
}
`;

exports[`Auth reducer should handle SET_FORM_ERROR_START 1`] = `
Object {
  "authRedirectPath": "/",
  "error": null,
  "loading": false,
  "registering": false,
  "registrationComplete": false,
  "token": null,
  "userId": null,
}
`;

exports[`Auth reducer should return the initial state 1`] = `
Object {
  "authRedirectPath": "/",
  "error": null,
  "loading": false,
  "registering": false,
  "registrationComplete": false,
  "token": null,
  "userId": null,
}
`;
